things/database.py

```diff
@@ -683,6 +685,30 @@ def make_filter(column, value):
     }.get(value, default)
 
 
+def make_date_filter(create_var_name, last):
+    """Returns SQL filter to limit date range of the SQL query.
+
+    Args:
+        create_var_name (string): Name of the column that has information on when each task was created.
+        last (string): A string comprised of an integer and a single character that can be 'd' or 'w' that determines whether to return all tasks for the past X days or weeks.
+
+    Returns:
+        string: A string to add into the filter for the SQL query.
+    """
+    if (type(last) is str):
+        if last[-1] == "d":
+            return f'AND datetime(TASK.{create_var_name}, "unixepoch", "localtime") BETWEEN datetime("now", "-{int(last[:-1])} days") AND datetime("now", "localtime")'
```

Comment:
You might simplify to:
```python
number = int(offset[:-1])

if suffix == "d":
    modifier = f"-{number} days"
elif suffix == "w":
    modifier = f"-{7 * number} days"
```
...

return sql filter:
```sql
datetime(TASK.{created_column}, "unixepoch", "localtime") > datetime("now", "{modifier}")
```

Reply:
Great suggestions. I added them in the most recent commit.